modules/signals/src/state-source.ts

```diff
@@ -94,6 +94,10 @@ export function patchState<State extends object>(
     }
     const signalKey = key as keyof State;
 
+    if (currentState[signalKey] === newState[signalKey]) {
```

Comment:
@markostanimirovic, @timdeschryver.

I had to re-introduce that check here.

Let's say, we have

```ts
withState({ id: 1 }),
withLinkedState( ({ id }) => {
  name: () => {
    id();
    return 'John';
  }
})
```

If I do a `patchState(store, { id: 2 })`, then `patchState` gets the current values of all properties. Once it updates `id`, it will reset synchronously `name`. The problem is now that `patchState` might also update `name` with its old value, although it already go a new name.